feat(archlinuxcn-mirrorlist): Update

Zenvie opened the pull request.

Zenvie committed:
refactor: rename files for clarity

Zenvie committed:
style: format with ruff and prettier

- Format the files using the default settings of ruff and prettier
- Validate that yaml file comply with the yamllint rules

Zenvie committed:
feat(manager.py)!: optimize code

- Follow ruff, pylint, pyupgrade, and refurb rules
- Modify README_ITEM_TEMPLATE and README_TEMPLATE templates to add more details in README; Validate that markdown file comply with the markdownlint-cli2 rules
- Modify MIRRORLIST_ITEM_TEMPLATE template and mirrorlist_item function to append comments for each mirror site in mirrorlist
- Use ruamel.yaml module to preserve yaml file content, eliminating the need for ordered_load/dump_yaml functions
- Remove unused mirror_score function
- Modify mirror_title, generate_readme, generate_mirrorlist functions, removing conditional statements
- Modify mirror_comment function, retaining only comments
- Modify generate_mirrorlist function, enhance the way of generating time
- Modify sub_list function, returning only mirror names
- Modify the feature of updating protocols
  - Divide functions, easy to maintain
  - Use httpx and socket module to determine whether the website supports http/https and ipv4/ipv6; Increase timeout when connecting; Output exceptions when connection fails
- Modify the feature of updating geographic coordinates
  - Break down functions, easy to maintain
  - Use httpx module that supports http/2 to structured query; Display information only when geographic location does not exist; Output exceptions when connection fails
  - Fix code logic, ensure that geographic coordinates data are all from openstreetmap
- Modify the feature of generating geojson file
  - Separate functions, easy to maintain
  - Fix code logic, ensure that the geographic location or coordinates are not lost
  - Use orjson module to generate
- Modify sub_all function, execute all operations
- Modify help document
  - Refine wording
  - Avoid force exit the program when entering wrong parameters
- Code has no log output feature, discard nicelogger module

Zenvie committed:
feat(mirrors): update information

- Update the geographic location of the CDN mirror sites[^1][^2]
- README.md, archlinuxcn-mirrorlist and mirrors.geojson are generated by manager.py

[^1]: https://www.tencentcloud.com/document/product/228/2941
[^2]: https://www.alibabacloud.com/help/en/cdn/product-overview/pop-distribution

Zenvie committed:
feat(stats.py): improve performance

- Follow ruff, pylint, pyupgrade, and refurb rules
- Use the ruamel.yaml module to load the yaml file
- Remove the humantime function that is no longer used
- Split the functions for easy maintenance
- Use the httpx module that supports http/2 to get the last modified time of multiple file
- Save result to json file

Zenvie committed:
feat(application.md): translate to English

- Validate that markdown file comply with the markdownlint-cli2 rules

Zenvie committed:
feat(map.html): enhance features

- Follow the prettier, stylelint-config-standard and stylelint-config-recess-order rules
- Change the page title; Add favicon.ico and page description
- Upgrade the dependencies to the latest versions
- Change the legend and popup style
- Set the zoom level of the map; Avoid showing the coverage area on mouse hover
- Remove scale bar and sorting mirror sites function
- Improve the color of the mirror site markers

Zenvie committed:
feat(list.html): add webpage

- Follow the prettier, stylelint-config-standard and stylelint-config-recess-order rules

auxiliarypower commented:
> I prefer to use single quotes when possible as they are easier to type.

This is the default configuration of Ruff.

> "Duesseldorf" is the ASCII transscript of "Düsseldorf". Maybe we should just use "Düsseldorf".

All the geographic location names are written as the titles of the Wikipedia articles.

auxiliarypower commented on manager.py:
```diff
@@ -126,280 +70,269 @@ def readme_item(item):
+    geojson = generate_geojson(mirrors)
+    with open(OUTPUT_GEOJSON, "w", encoding="utf-8") as output:
+        print(
+            re.sub(
+                r"(\[\n\s+)(.*)(,\n\s+)(.*)(\n\s+\])",
+                r"[\2, \4]",
```
Follow prettier rules.

auxiliarypower commented on archlinuxcn-mirrorlist:
```diff
@@ -1,106 +1,112 @@
+## Our main server (Amsterdam, North Holland, Netherlands) (http, https, ipv4, ipv6)
 # Server = https://repo.archlinuxcn.org/$arch
 
-## CERNET (中国) (ipv4, ipv6, http, https)
+## 校园网联合镜像站 (中国) (ipv4, ipv6)
+## Redirect to suitable educational mirror based on location
```
Since this mirror site does not have a real path, all the supported protocols work as expected.

auxiliarypower commented on mirrors.yaml:
```diff
@@ -2,705 +2,780 @@
+      - New York, United States
+      - San Jose, United States
+      - Santa Clara, United States
+      - Seattle, United States
```
All data is sourced from official documents, with only the names of places corrected to match the title of Wikipedia entries.

auxiliarypower replied:
Please refer to the commit message. I only changed the way of writing place names.

auxiliarypower commented on status.py:
```diff
@@ -1,74 +1,114 @@
+        headers={"User-Agent": "httpx/0.25.2"},
+        http2=True,
+        timeout=10,
+        follow_redirects=True,
```
Without this parameter, the CERNET mirror site cannot be tested, although testing this mirror site is meaningless.

auxiliarypower commented on manager.py:
```diff
@@ -126,280 +70,269 @@ def readme_item(item):
-
+def sub_protocols(_):
+    mirrors = []
+    yaml = ruamel.yaml.YAML()
+    with open(SOURCE_YAML, encoding="utf-8") as source:
+        try:
+            mirrors = yaml.load(source)["archlinuxcn"]
+        except ruamel.yaml.YAMLError as error:
+            sys.exit(repr(error))
+    update_protocols(mirrors)
+    with open(SOURCE_YAML, "w", encoding="utf-8") as output:
+        yaml.explicit_start = True
+        yaml.indent(mapping=2, sequence=4, offset=2)
+        yaml.dump({"archlinuxcn": mirrors}, output)
```
To preserve the original order when writing to yaml file, do not add the (typ="safe") argument.

auxiliarypower commented on README.md:
````diff
@@ -1,256 +1,304 @@
 ```ini
-## CERNET (中国) (ipv4, ipv6, http, https)
-## Added: 2023-08-19
-## This will redirect you to the nearest China educational mirror site. However, please note that some sites may not support IPv6.
+## 校园网联合镜像站 (中国) (ipv4, ipv6)
+## Redirect to suitable educational mirror based on location
 [archlinuxcn]
 Server = https://mirrors.cernet.edu.cn/archlinuxcn/$arch
 ```
````
This is an automatically redirected mirror site, so it is impossible to get a status code of 200 as a response.

auxiliarypower commented:
According to the license information of [PKGBUILD](https://gitlab.archlinux.org/archlinux/packaging/packages/pacman-mirrorlist/-/blob/main/PKGBUILD?ref_type=heads#L10), we should consider adopting the GPLv3 license. For more details on license compatibility, please refer to the websites of [the OSM Foundation License](https://osmfoundation.org/wiki/Licence/Licence_and_Legal_FAQ#1.5._Is_OSM_Data_Free_to_Use?) and [the Creative Commons license](https://wiki.creativecommons.org/wiki/ShareAlike_compatibility:_GPLv3#Considerations_for_adapters_applying_the_GPLv3).

auxiliarypower commented on manager.py:
```diff
@@ -126,280 +70,269 @@ def readme_item(item):
+        except ruamel.yaml.YAMLError as error:
+            sys.exit(repr(error))
+    geojson = generate_geojson(mirrors)
+    with open(OUTPUT_GEOJSON, "w", encoding="utf-8") as output:
+        print(
+            re.sub(
+                r"(\[\n\s+)(.*)(,\n\s+)(.*)(\n\s+\])",
+                r"[\2, \4]",
```
In order to format the generated geojson file according to Prettier’s formatting standards.

auxiliarypower commented on status.py:
```diff
@@ -1,75 +1,116 @@
+            "url": "https://repo.archlinuxcn.org/",
+        }
+    )
+    client = httpx.AsyncClient(
+        headers={"User-Agent": "httpx/0.25.2"},
+        http2=True,
+        timeout=10,
+        follow_redirects=True,
+    )
```
httpx will use http/1.1 for mirror sites that do not support http/2.

Although it is very difficult to test CERNET mirror site without the parameter (follow_redirects=True), it is pointless to do so.

Zenvie committed:
feat(mirrors): add sau and jcut

- README.md, archlinuxcn-mirrorlist and mirrors.geojson are generated by manager.py

Zenvie committed:
feat(mirrors)!: rearrange the mirrors

- Follow ruff, pylint, pyupgrade, and refurb rules
- README.md, archlinuxcn-mirrorlist and mirrors.geojson are generated by manager.py

Wonderwhothen commented:
[](url) application was not available  
There is a  file that is incomplete .
There still a conflict.  Complete  in  the
 next six hours.    Thank you.
